diff --git a/include/argparse/argparse.hpp b/include/argparse/argparse.hpp
index 06d30fd..43af72d 100644
--- a/include/argparse/argparse.hpp
+++ b/include/argparse/argparse.hpp
@@ -1964,6 +1964,14 @@ public:
     return is_subcommand_used(subparser.m_program_name);
   }
 
+  /*
+   * temporary solution
+   * Getter that returns program name
+   */
+  auto get_program_name () const {
+    return m_program_name;
+  }
+  
   /* Indexing operator. Return a reference to an Argument object
    * Used in conjunction with Argument.operator== e.g., parser["foo"] == true
    * @throws std::logic_error in case of an invalid argument name
